lib/ansible/modules/cloud/amazon/iam_managed_policy.py

```diff
@@ -0,0 +1,326 @@
+
+# Update a policy with a new default version
+- name: Create IAM Managed Policy
+  iam_managed_policy:
+    policy_name: "ManagedPolicy"
+    policy_document: managed_policy_update.json
+    state: present
+
+# Update a policy with a new non default version
+- name: Create IAM Managed Policy
+  iam_managed_policy:
+    policy_name: "ManagedPolicy"
+    policy_document: managed_policy_update.json
+    make_default: false
+    state: present
+
+# Update a policy and make it the only version and the default version
+- name: Create IAM Managed Policy
+  iam_managed_policy:
+    policy_name: "ManagedPolicy"
+    policy_document: managed_policy_update.json
+    only_version: true
+    state: present
+
+# Remove a policy
+- name: Create IAM Managed Policy
+  iam_managed_policy:
+    policy_name: "ManagedPolicy"
+    state: absent
+'''
+
+RETURN = '''
+policy:
+  description: Returns the policy json structure, when state == absent this will return the value of the removed policy.
+  returned: success
+  type: string
+  sample: '{
+        "Arn": "arn:aws-us-gov:iam::268171240745:policy/AmazonS3IGAccess",
+        "AttachmentCount": 0,
+        "CreateDate": "2017-03-01T15:42:55.981000+00:00",
+        "DefaultVersionId": "v1",
+        "IsAttachable": true,
+        "Path": "/",
+        "PolicyId": "ANPALM4KLDMTFXGOOJIHG",
+        "PolicyName": "AmazonS3IGAccess",
+        "UpdateDate": "2017-03-01T15:42:55.981000+00:00"
+  }'
+'''
+from ansible.module_utils.basic import AnsibleModule
+import ansible.module_utils.ec2
+
+import json
+
+try:
+    import boto3
+    import botocore
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def get_policy_by_name(iam, name, **kwargs):
+    response = iam.list_policies(Scope='Local', **kwargs)
+    for policy in response['Policies']:
+        if policy['PolicyName'] == name:
+            return policy
+    if response['IsTruncated']:
+        return get_policy_by_name(iam, name, marker=response['marker'])
+    return None
+
+
+def delete_oldest_non_default_version(iam, policy):
+    versions = [v for v in iam.list_policy_versions(PolicyArn=policy['Arn'])[
+        'Versions'] if not v['IsDefaultVersion']]
+    versions.sort(key=lambda v: v['CreateDate'], reverse=True)
+    for v in versions[-1:]:
+        iam.delete_policy_version(
+            PolicyArn=policy['Arn'],
+            VersionId=v['VersionId']
+        )
+
+# This needs to return policy_version, changed
+
+
+def get_or_create_policy_version(iam, policy, policy_document):
+    versions = iam.list_policy_versions(PolicyArn=policy['Arn'])['Versions']
+    for v in versions:
+        document = iam.get_policy_version(
+            PolicyArn=policy['Arn'],
+            VersionId=v['VersionId'])['PolicyVersion']['Document']
+        if json.dumps(
+                document,
+                sort_keys=True) == json.dumps(
+                json.loads(policy_document),
+                sort_keys=True):
+            return v, False
```

Comment:
Use the helper function in module utils for comparison. See AWS guidelines for details.